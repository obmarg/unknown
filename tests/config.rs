// TODO: At some point use libtest-mimic to make this smarter?  CBA now.
use assert_cmd::Command;

#[test]
fn task_require_non_ancestor() {
    test_failing_config("task_require_non_ancestor");
}

#[test]
fn task_require_unknown_project() {
    test_failing_config("task_require_unknown_project");
}

fn test_failing_config(name: &str) {
    let mut cmd = Command::cargo_bin("unknown").unwrap();
    cmd.arg("projects");
    cmd.current_dir(format!("tests/config-test-cases/{name}"));

    let assert = cmd.assert().failure();
    let stdout = String::from_utf8_lossy(&assert.get_output().stdout);
    let stderr = String::from_utf8_lossy(&assert.get_output().stderr);

    insta::assert_snapshot!(format!("{name}.stdout"), stdout.as_ref());
    insta::assert_snapshot!(format!("{name}.stderr"), stderr.as_ref());
}
