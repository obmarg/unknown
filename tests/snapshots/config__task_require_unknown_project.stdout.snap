---
source: tests/config.rs
expression: stdout.as_ref()
---

